src/rgw/rgw_process.cc

```diff
@@ -327,18 +328,29 @@ int process_request(const RGWProcessEnv& penv,
     abort_early(s, NULL, -ERR_METHOD_NOT_ALLOWED, handler, yield);
     goto done;
   }
+  is_health_or_system_request = (
+    (op->get_type() == RGW_OP_GET_HEALTH_CHECK) || s->user->get_info().admin ||
+    s->user->get_info().system);
```

Comment:
what's the motivation for checking for `admin`/`system` users? those can issue s3 requests like normal users

Reply:
> what's the motivation for checking for `admin`/`system` users? those can issue s3 requests like normal users

the same as for healthcheck, why do admin requests need to check for lua script ? if you want to debug the s3 request, can always send the s3 request to debug using lua ?

Reply:
i assume that admin/system requests are emitted by the RGW itself?

Reply:
admin and system are just flags on the user info. nothing prevents s3 clients from issuing requests as those users, and rgw should handle them the same as normal users (except for the permission checks)

its common for cluster admins to have an admin user in case they need to fix broken acls/policies via s3 apis, or interact with the /admin apis. if there are lua scripts in place for any of these actions, i think they should apply to admin/system requests too

Reply:
done.